libraries/base/Data/Semigroup.hs

```diff
@@ -18,21 +18,36 @@
 -- Maintainer  :  libraries@haskell.org
 -- Stability   :  provisional
 -- Portability :  portable
+-- 
+-- A type @a@ is a 'Semigroup' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one. 
```

Comment:
It would be good to define "associative" here. Perhaps something like,
```
Being associative, the function must obey the following law,
>>> (a <> b) <> c  ==  a <> (b <> c)
```